File: frontend/services/habitService.js

import axios from 'axios';

class HabitService {
  static addStatus(date, habit, done) {
    return axios.post('/habit', { date, habit, done });
  }
  static getStatus(date) {
    return axios.get(`/habit?date=${date}`);
  }
  static getProgress(week_start) {
    return axios.get(`/habit/progress?week_start=${week_start}`);
  }
}

export default HabitService;

File: src/service/habit_service.py

from datetime import datetime, timedelta

class HabitService:
    def __init__(self):
        # store entries as date_str -> {habit: bool}
        self.entries = {}

    def save_status(self, date_str, habit, done):
        day = self.entries.setdefault(date_str, {})
        day[habit] = done
        return {'date': date_str, 'habit': habit, 'done': done}

    def get_status(self, date_str):
        return self.entries.get(date_str, {})

    def get_weekly_progress(self, week_start):
        # week_start: 'YYYY-MM-DD'
        start = datetime.fromisoformat(week_start)
        progress = {}
        for i in range(7):
            d = (start + timedelta(days=i)).date().isoformat()
            progress[d] = self.entries.get(d, {})
        return progress

# singleton
habit_service = HabitService()
